### PR TITLE
Add multi-return for snow simulation

### PR DESCRIPTION
**Pull request by PawelLiberadzki**
Description

This PR introduces two things (based on updates to RGL library):
- new implementation of LiDAR return modes - there are more return modes available and they are independent of laser beam divergence,
- update to snow simulation for multi-return feature - now snow simulation works correctly with updated return modes and divergence; what is also important, snow simulation, return modes and beam divergence are independent on each other, e.g. user may decide to turn on snow simulation, choose single return mode 'first', and turn off beam divergence - each setup should work correctly.

### CODE REVIEW

File: Assets/RGLUnityPlugin/Scripts/LidarSensor.cs
Comment (msz-rai) [inline Assets/RGLUnityPlugin/Scripts/LidarSensor.cs:62]: The default value should be decided.

**Review by msz-rai — COMMENTED** [review]
The code looks good :+1: 
I am waiting for docs update to approve it.

File: Assets/RGLUnityPlugin/Scripts/LidarSensor.cs
Comment (PawelLiberadzki) [reply Assets/RGLUnityPlugin/Scripts/LidarSensor.cs:62]: For the feature branch, I set this to false - this is exactly the approach that was used before. Final decision will be made before merge to main branch.

Comment (msz-rai) [inline Assets/RGLUnityPlugin/Scripts/LidarSensor.cs:62]:
```suggestion
        public bool simulateBeamDivergence = false;
```

**Review by msz-rai — APPROVED** [review]
:+1:

File: Assets/RGLUnityPlugin/Scripts/LidarSnowManager.cs
Comment (msz-rai) [inline Assets/RGLUnityPlugin/Scripts/LidarSnowManager.cs:60]: To be removed.

**Pull request merged**
